api/v1alpha3/azuremachine_types.go

```diff
@@ -34,7 +34,13 @@ type AzureMachineSpec struct {
 	// +optional
 	ProviderID *string `json:"providerID,omitempty"`
 
-	VMSize           string           `json:"vmSize"`
+	VMSize string `json:"vmSize"`
+
+	// FailureDomain is the failure domain unique identifier this Machine should be attached to,
+	// as defined in Cluster API. This relates to an Azure Availability Zone
+	FailureDomain *string `json:"failureDomain,omitempty"`
+
+	// DEPRECATED: use FailureDomain instead
```

Comment:
would it make sense to print a warning if `availabilityZone` is used?

Reply:
Good idea, i've added a warning.